src/rard/research/models/text_object_field.py

```diff
@@ -51,6 +53,12 @@ def update_mentions(self):
             # get related queryset
             class_mentions_queryset = getattr(self, class_name + "_mentions")
             for instance in class_mentions_queryset.all():
+                try:
+                    get_object_or_404(instance.__class__, pk=instance.pk)
+                except Http404:
+                    instance.mentioned_in.remove(
+                        self
+                    )  # if instance no longer exists, remove it from mentioned_in
```

Comment:
A queryset wouldn't contain objects that don't exist. I don't understand why this would be necessary. You've got a reference to an instance and then using that instance's properties to look it up, so how would that ever result in it not being found?

Reply:
if an object that's been deleted is mentioned, I thought it's still in the content and only in the non editing view (TOF preview) is it changed to `bad-link`? I ran into this issue previously but possibly the full flow fixes that?

Reply:
ah - calling the `update_content_mentions` should handle that so it shouldn't be an issue anymore